Rewrite website using VuePress

alessandrojean opened the pull request:
**Edit:** I just read about the part of creating an Issue first to discuss major changes now, so sorry if this PR was too sudden. As I said, the initial goal is only to be a suggestion. Feel free to close this PR if it's better.

*This PR is just a suggestion to improve the development process of the website. It's okay if it's not merged and I am open to newer suggestions.*

I used [VuePress](https://vuepress.vuejs.org) to rewrite the website, so now the content can be written using Markdown, allowing translations to other languages, making the content accessible to more people.

Since the previous version uses NPM too for the build process, I only updated the commands to the VuePress's ones in the `package.json`. VuePress will build the website in the same folder (`./public`), so the GitHub build process should work the same way, but this part I can't test unfortunately.

To test locally, just run:

```console
$ npm install
$ npm run serve
```

A gallery of images is available [here](https://imgur.com/a/DeibuiS).

arkon commented:
VuePress or Gatsby are definitely appealing to me since they're in the NodeJS ecosystem. I did write the original npm-run-scripts with a drop-in replacement like this in mind so the actions workflow would still work the same (assuming the build artifacts end up in the same place).

Do you have any thoughts on VuePress over other alternatives? You didn't seem to explain why it was chosen.

I didn't do a deep dive into the changes, but it does look like it contains all the existing content so that's good. One thing I didn't like is how internationalization is handled -- it seems a bit clunky. I guess the need for internationalization is up for debate too, since it's a maintenance effort that I'm not sure will age nicely.

alessandrojean commented:
> Do you have any thoughts on VuePress over other alternatives? You didn't seem to explain why it was chosen.

I have worked with both frameworks that VuePress and Gatsby use (Vue.js and React, respectively), and what I noticed is that the learning curve of React is more difficult, comparing to the Vue one. Creating components and working of the reactivity is much simpler with Vue. This is why I chose VuePress, I think it will be more easier to the people contribute to the repository in the future, focusing on the content.

> One thing I didn't like is how internationalization is handled -- it seems a bit clunky.

I didn't feel it when developing, because since the content is written in Markdown, I can't think of solutions that are very different from what they use actually.

> I guess the need for internationalization is up for debate too, since it's a maintenance effort that I'm not sure will age nicely.

You are right. Every update in the main page, the English one, will make the other translated pages outdated until someone works on them.

What I can suggest is initially remove the internationalization, and focus only on the transition to the VuePress system. Since VuePress supports the internationalization natively, it's easy to open a separate debate in the future and implement if it's decided to.

alessandrojean commented:
I removed the initial internationalization for now and added an extension list that uses the JSON available in the extension repo. The list looks like [this](https://imgur.com/UsDABmU).

arkon commented on package.json:
```diff
@@ -2,15 +2,21 @@
 	"private": true,
 	"devDependencies": {
-		"http-server": "^0.11.1"
+		"http-server": "^0.11.1",
```
Can remove this.

arkon commented on package.json:
```diff
@@ -2,15 +2,21 @@
+		"vuepress": "^1.1.0"
+	},
+	"dependencies": {
+		"axios": "^0.19.0",
```
Why axios over a fetch polyfill or something?

alessandrojean replied:
Just a personal choice.

arkon commented on package.json:
```diff
@@ -2,15 +2,21 @@
+	"dependencies": {
+		"axios": "^0.19.0",
+		"iso-639-1": "^2.1.0",
+		"lodash": "^4.17.15"
```
Can we pull in the individual packages?

alessandrojean replied:
I think I didn't understand very well, are you talking about the Lodash methods?

arkon replied:
Yeah, e.g. https://www.npmjs.com/package/lodash.groupby

arkon commented on src/.vuepress/components/ExtensionList.vue:
```diff
@@ -0,0 +1,104 @@
+</template>
+
+<script>
+const axios = require('axios')
```
I personally belong to the church of having semicolons.

Another thing -- can we use ES2015 imports here, considering we're using ES2015 exports below?

alessandrojean replied:
Don't worry, I can put the semicolons, haha.
And yes, I will change to use the `import` statement.

arkon commented on src/.vuepress/components/ExtensionList.vue:
```diff
@@ -0,0 +1,104 @@
+}
+
+.extension .button {
+    display: inline-block;
```
Some mixed indentation here.

arkon commented on src/.vuepress/styles/index.styl:
```diff
@@ -0,0 +1,5 @@
+.centered
```
Can we use regular CSS, or maybe SCSS? Feels weird to mix Stylus and regular CSS in the Vue files.

alessandrojean replied:
I think VuePress uses Stylus and this can't be changed inside the `styles` folder, as the documentation says that to customize the style you have to create the `index.styl` and the other file. I will see if I can make the components use Stylus too to standardize.

arkon replied:
https://code.luasoftware.com/tutorials/vuepress/vuepress-css-or-sass/

;)

arkon commented on src/README.md:
```diff
@@ -0,0 +1,17 @@
+      details: MyAnimeList, AniList and Kitsu support.
+    - title: Configurable
+      details: Reader with multiple viewers, reading directions and other settings.
+footer: Apache Licensed | Copyright © 2015-present Javier Tomás
```
Should update this.

alessandrojean replied:
To what, exactly?

arkon replied:
Actually nevermind, that already matches the LICENSE for the app. Mistake on my part.

arkon commented on src/faq/README.md:
```diff
@@ -0,0 +1,135 @@
+
+You can find more frequently asked questions [here].
+
+[here]: https://github.com/inorichi/tachiyomi/wiki/FAQ#frequently-asked-questions
```
Don't really see the point of keeping this separate. Same for other instances.

alessandrojean replied:
Just my personal style of writing Markdown, I can change this.

alessandrojean commented:
I changed the suggestions you made, with the exception of the file `palette.styl` because VuePress uses it to generate the default theme, that internally uses Stylus.

arkon commented:
And it's live: https://tachiyomi.org

Looks good! :)
